### DISCUSSION TITLE
when running gocd analytics plugin, what would be easiest way to get alerts if a pipeline build time increased more than X%

### DISCUSSION THREAD

**Discussion by pracplayopen** [gocd/gocd#13123] [Q&A]
would this require us to write a script or similiar that interacted with directly w/postgres?

i'm assuming there is no way to do this w/plugin or gocd configuration directly.
didn't see any plugin that offered such a thing (presume it would depend on using analytics plugin)

not sure what the simplest way to do this would be.  it seems like it should be conceptually simple looking at data offered via analytics plugin UI:

eg:   

```

if GetPipelineBuildTime(<pipeline>) >  1.1 * GetPipelineBuildTimes(<pipeline>).Last(5).Avg() then sendnotification(<pipeline>,"Pipeline is unexpectedly slow");

```

Any suggestions appreciated (even if the simplest way does not look like this)

thx

**Comment by sriramnrn** [comment]
On Mon, Sep 30, 2024 at 9:58 PM pracplay open code ***@***.***>
wrote:

> would this require us to write a script or similiar that interacted with
> directly w/postgres?
>
> i'm assuming there is no way to do this w/plugin or gocd configuration
> directly.
> didn't see any plugin that offered such a thing (presume it would depend
> on using analytics plugin)
>
> not sure what the simplest way to do this would be. it seems like it
> should be conceptually simple looking at data offered via analytics plugin
> UI:
>
eg:
>
>
> if GetPipelineBuildTime(<pipeline>) >  1.1 * GetPipelineBuildTimes(<pipeline>).Last(5).Avg() then sendnotification(<pipeline>,"Pipeline is unexpectedly slow");
>
>
> Any suggestions appreciated (even if the simplest way does not look like
> this)
>
The current Analytics plugin requires you to get the highcharts library
independently. Some of us are working on a revamp of the plugin for
pipeline analysis, and highlighting an aberration (higher or lower time) is
on the roadmap but we've not got to it yet ( We hope to announce the plugin
by end-October).

For what you are trying to achieve, a poll based on your formula would
suffice, I think.

**Comment by chadlwilson** [comment]
> when running gocd analytics plugin, what would be easiest way to get alerts if a pipeline build time increased more than X%
> 
> would this require us to write a script or similiar that interacted with directly w/postgres?
> 
> i'm assuming there is no way to do this w/plugin or gocd configuration directly. didn't see any plugin that offered such a thing (presume it would depend on using analytics plugin)

Plugins don't offer web APIs on their own, so if there is preprocessed data you'd like access to, which is specific to the analytics plugin database you'd need to use the database directly - yes.

Having said this, I am not familiar with the analytics plugin schema and whether it pre-calculates answers to questions in a way useful to you. The alternative for some simpler questions may be to interact with the GoCD APIs directly.
